### ISSUE TITLE
Freebsd build failing; regression

### ISSUE THREAD

**Issue by easytarget** [micropython/micropython#15430]
### Port, board and/or hardware

unix

### MicroPython version

Micropython 1.23.0 on FreeBSD14.1

### Reproduction

Build failure on FreeBSD 14.1
Cause is as per: #1972 (conflict of `re_exec` with freebsd uisntd.h)

Commit 7f5d5c72 appears to revert the fix for the above in `extmod/modure.c` as part of a mass-rename of some modules and functions.

https://github.com/micropython/micropython/commit/7f5d5c72718af773db751269c6ae14037b9c0727

### Expected behaviour

A working build ;-)

### Observed behaviour

```make
../../extmod/modre.c:197:17: error: conflicting types for 're_exec'; have 'void *(_Bool,  uint,  void * const*)' {aka 'void
 *(_Bool,  unsigned int,  void * const*)'}
  197 | static mp_obj_t re_exec(bool is_anchored, uint n_args, const mp_obj_t *args) {
      |                 ^~~~~~~
In file included from ./mpconfigport.h:33,
                 from ../../py/mpconfig.h:91,
                 from ../../py/mpstate.h:31,
                 from ../../py/runtime.h:29,
                 from ../../extmod/modre.c:31:
/usr/include/unistd.h:554:10: note: previous declaration of 're_exec' with type 'int(const char *)'
  554 | int      re_exec(const char *);
      |          ^~~~~~~
See https://github.com/micropython/micropython/wiki/Build-Troubleshooting
gmake: *** [../../py/mkrules.mk:90: build-standard/extmod/modre.o] Error 1
```

### Additional Information

No, I've provided everything above.

### Code of Conduct

Yes, I agree

**Comment by easytarget** [comment]
Manually re-imposing the changes (renaming to `ure_exec` in `modre.c`) allows a successful build.

**Comment by dlech** [comment]
I would suggest submitting a pull request that adds a `mp_` prefix instead to avoid the collision.

**Comment by dpgeorge** [comment]
Fixed by 4fdad8eabef17aa8b920484e0af7db5ad9889167
